tests/foreman/api/test_template.py

```diff
@@ -42,303 +44,232 @@
-        new_name = gen_string('alpha')
-        org = entities.Organization().create()
-        loc = entities.Location().create()
-        template = entities.ProvisioningTemplate(organization=[org], location=[loc]).create()
-        # Create user with Manager role
-        role = entities.Role().search(query={'search': 'name="Manager"'})[0]
-        entities.User(
-            role=[role],
-            admin=False,
-            login=user_login,
-            password=user_password,
-            organization=[org],
-            location=[loc],
+        template = entities.ProvisioningTemplate(
+            name=name,
+            organization=[module_org],
+            location=[module_location],
+            snippet=False,
+            template_kind=template_kind,
         ).create()
-        # Update template name with that user
-        cfg = get_nailgun_config()
-        cfg.auth = (user_login, user_password)
+        assert template.name == name
+        assert len(template.organization) == 1, "Template should be assigned to a single org here"
+        assert template.organization[0].id == module_org.id
+        assert len(template.location) == 1, "Template should be assigned to a single location here"
+        assert template.location[0].id == module_location.id
+        assert template.snippet is False, "Template snippet attribute is True instead of False"
+        assert template.template_kind.id == template_kind.id
+
+        # negative create
+        with pytest.raises(HTTPError) as e1:
+            entities.ProvisioningTemplate(name=gen_choice(invalid_names_list())).create()
+        assert e1.value.response.status_code == 422
+
+        invalid = entities.ProvisioningTemplate(snippet=False)
+        invalid.create_missing()
+        invalid.template_kind = None
+        invalid.template_kind_name = gen_string('alpha')
+        with pytest.raises(HTTPError) as e2:
+            invalid.create(create_missing=False)
+        assert e2.value.response.status_code == 422
+
+        # update
+        assert template.template_kind.id == template_kind.id, "Template kind id doesn't match"
         updated = entities.ProvisioningTemplate(cfg, id=template.id, name=new_name).update(
             ['name']
         )
-        self.assertEqual(updated.name, new_name)
-
-    @tier1
-    def test_negative_update_name(self):
-        """Create configuration template then update its name to an
-        invalid name.
-
-        :id: f6167dc5-26ba-46d7-b61f-14c290d6a8fa
-
-        :expectedresults: Configuration Template is created, and its name is
-            not updated.
-
-        :CaseImportance: Medium
+        assert updated.name == new_name, "The Provisioning template wasn't properly renamed"
+        # clone
+        template_origin = template.read_json()
+        # remove unique keys
+        unique_keys = ('updated_at', 'created_at', 'id', 'name')
+        for key in unique_keys:
+            del template_origin[key]
```

Comment:
dict comprehension?
```python
template_origin = template.read_json()
unique_keys = ('updated_at', 'created_at', 'id', 'name')
template_origin = {key: value for key, value in template_origin.items() if key not in unique_keys}
```
Dunno, feels more pythonic.

Reply:
you are right, this was the original code structure and i didn't update it. I will do it now